### ISSUE TITLE
Cannot remove product_list_toolbar in XML

### ISSUE THREAD

**Issue by joshfortyfour** [magento/magento2#9413]
### Preconditions

1. Magento 2.1.6
2. PHP 7.0.18
3. MySql 5.7.18
4. Apache 2.4

### Steps to reproduce

1. Create Magento_Catalog/layout/catalog_category_view.xml in custom theme
2. Add the following code:
```
<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
        <referenceContainer name="content">
            <referenceBlock name="product_list_toolbar" remove="true"/>
        </referenceContainer>
    </body>
</page>
```

### Expected result

1. The toolbar should be removed from the product list pages.

### Actual result

1. The toolbar remains on the page. After talking with Alan Storm about the issue he dug into the `vendor/magento/module-catalog/Block/Product/ListProduct.php` file and, specifically, the `getToolbarBlock()` method which does this:

```
public function getToolbarBlock()
    {
        $blockName = $this->getToolbarBlockName();
        if ($blockName) {
            $block = $this->getLayout()->getBlock($blockName);
            if ($block) {
                return $block;
            }
        }
        $block = $this->getLayout()->createBlock($this->_defaultToolbarBlock, uniqid(microtime()));
        return $block;
    }
```

It checks for the toolbar and, if it finds it, returns it. If it doesn't find it (i.e., it's removed via xml), it renders it to the page.

**Comment by tuyennn** [comment]
See this topic
[http://magento-quickies.alanstorm.com/post/160022140410/why-you-cant-remove-the-productlisttoolbar
](url)


This is why you cannot remove product_list_toolbar

**Comment by astorm** [comment]
@tuyennn Based on @joshfortyfour's bug report (including mentioning the `getToolbarBlock`) it sounds like he's already aware of the underlying reasons the toolbar block can't be removed,and considers those underlying reasons a bug.

**Comment by tuyennn** [comment]
@astorm Yeah, also your topic is quite well. I think your explanation should help to debug and resolve this one.

**Comment by VincentMarmiesse** [comment]
Hello @tuyennn ,
I'm facing the same issue and as @astorm mentioned, I understand why the `product_list_toolbar` can't be removed with just a `remove="true"`.

But it's a bug IMO and I think it should be fixed in the future 👍

**Comment by magento-engcom-team** [comment]
@joshfortyfour, thank you for your report.
We've created internal ticket(s) MAGETWO-80293 to track progress on the issue.

**Comment by mariuscris** [comment]
I'm working on it #SQUASHTOBERFEST

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop branch. Will be available with upcoming patch release

**Comment by VincentMarmiesse** [comment]
Hi @okorshenko,

Is it supposed to be fixed in 2.2.2?

I'm having the issue on a 2.2.2.

**Comment by okorshenko** [comment]
I think it will be a part of 2.2.4

**Comment by okorshenko** [comment]
Hi @joshfortyfour. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1380 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - [e9238bcb04f75a5ef0d801ef7a4c372832d83aad](https://github.com/magento/magento2/commit/e9238bcb04f75a5ef0d801ef7a4c372832d83aad)
 - [c46acb7f26afb967c2e727d9557d00a03418a599](https://github.com/magento/magento2/commit/c46acb7f26afb967c2e727d9557d00a03418a599)
 - [3d5cc519f30a8664aadca0b1378ff62138fd1bf9](https://github.com/magento/magento2/commit/3d5cc519f30a8664aadca0b1378ff62138fd1bf9)
 - [fcd3e7af7984ed9887a8213c51ff3cc29546fd93](https://github.com/magento/magento2/commit/fcd3e7af7984ed9887a8213c51ff3cc29546fd93)
 - [a37313a401cd143fbfb6642108f602a16d96d211](https://github.com/magento/magento2/commit/a37313a401cd143fbfb6642108f602a16d96d211)
 - [4e00485ed12b762e38bbf5e38cce97981a43dedf](https://github.com/magento/magento2/commit/4e00485ed12b762e38bbf5e38cce97981a43dedf)
 - [d73cca2aee6113f66e8098512343d4e440afe65e](https://github.com/magento/magento2/commit/d73cca2aee6113f66e8098512343d4e440afe65e)
 - [0d8938cfd7f6c85146e1ad3176947eb6f8a945ae](https://github.com/magento/magento2/commit/0d8938cfd7f6c85146e1ad3176947eb6f8a945ae)
 - [d544ca2a6d99132e66e8b416ab73cc712de149f9](https://github.com/magento/magento2/commit/d544ca2a6d99132e66e8b416ab73cc712de149f9)
 - [e075ca794af8189260359434488029b2768fc29d](https://github.com/magento/magento2/commit/e075ca794af8189260359434488029b2768fc29d)
 - [a6c2bc2b471586eb44d9b4fc2341517bb64cad15](https://github.com/magento/magento2/commit/a6c2bc2b471586eb44d9b4fc2341517bb64cad15)
 - [960d58cfc3ef200654f1c19a5eca8cb905da686c](https://github.com/magento/magento2/commit/960d58cfc3ef200654f1c19a5eca8cb905da686c)
 - [dbdb772c0260347267701f5f54196fec1c0e5109](https://github.com/magento/magento2/commit/dbdb772c0260347267701f5f54196fec1c0e5109)
 - [bac46ce7f62207e27fdb401e4e72040c2bed32db](https://github.com/magento/magento2/commit/bac46ce7f62207e27fdb401e4e72040c2bed32db)
 - [1503f3a49de2d20ed3729823a4410cbfba3fd92a](https://github.com/magento/magento2/commit/1503f3a49de2d20ed3729823a4410cbfba3fd92a)
 - [17662c6eb0948da16b97a93ad0c231947eba933d](https://github.com/magento/magento2/commit/17662c6eb0948da16b97a93ad0c231947eba933d)
 - [2a72e5cc81c7b28aa04c5c64a63a8493043d6384](https://github.com/magento/magento2/commit/2a72e5cc81c7b28aa04c5c64a63a8493043d6384)
 - [ddaf5c6df7e2b294085ef6133820a6a1b6b12762](https://github.com/magento/magento2/commit/ddaf5c6df7e2b294085ef6133820a6a1b6b12762)
 - [7bf80fdc06c32325040dada5082b78975b910e1b](https://github.com/magento/magento2/commit/7bf80fdc06c32325040dada5082b78975b910e1b)
 - [efb8bf6a94a99130fe013f4ef9c3c0c39777b7b1](https://github.com/magento/magento2/commit/efb8bf6a94a99130fe013f4ef9c3c0c39777b7b1)
 - [321263009819a694221ce8a06902cf7a4ba3ff00](https://github.com/magento/magento2/commit/321263009819a694221ce8a06902cf7a4ba3ff00)
 - [ac9c10b96940ed5393581135146c8ea29eddd143](https://github.com/magento/magento2/commit/ac9c10b96940ed5393581135146c8ea29eddd143)
 - [2c6c9c06b1a69a20f4b7bfcd27e1eebf8e26ad2b](https://github.com/magento/magento2/commit/2c6c9c06b1a69a20f4b7bfcd27e1eebf8e26ad2b)
 - [73522bf72d57e503281077abb20b4000375eb57d](https://github.com/magento/magento2/commit/73522bf72d57e503281077abb20b4000375eb57d)
 - [818cf61d28821cc2a5435c60b6308adcb5a3ea26](https://github.com/magento/magento2/commit/818cf61d28821cc2a5435c60b6308adcb5a3ea26)
 - [61d0d9608493f65a0c0eafe264e1575eaa84b7be](https://github.com/magento/magento2/commit/61d0d9608493f65a0c0eafe264e1575eaa84b7be)
 - [85e0956b08923ca721e3577edd684aec02a93b6b](https://github.com/magento/magento2/commit/85e0956b08923ca721e3577edd684aec02a93b6b)
 - [1698afc93aed22284efa2bf9d943b2108b5af3ab](https://github.com/magento/magento2/commit/1698afc93aed22284efa2bf9d943b2108b5af3ab)
 - [ce7faf6203e39aefdf177e2a6a4168cbb5223993](https://github.com/magento/magento2/commit/ce7faf6203e39aefdf177e2a6a4168cbb5223993)
 - [70051020df3e9053272cc6926680895b9898a35b](https://github.com/magento/magento2/commit/70051020df3e9053272cc6926680895b9898a35b)
 - [bd07bcc9e681810ac35fd59b3ef74970e55dca37](https://github.com/magento/magento2/commit/bd07bcc9e681810ac35fd59b3ef74970e55dca37)
 - [16930193a78f00b50f9fa16012d96c5c1d481f19](https://github.com/magento/magento2/commit/16930193a78f00b50f9fa16012d96c5c1d481f19)
 - [ba7da597361305e6a91c04fe39d63162408af3b3](https://github.com/magento/magento2/commit/ba7da597361305e6a91c04fe39d63162408af3b3)
 - [d9e8b87af003d550f39e9d34cf19f94088660a38](https://github.com/magento/magento2/commit/d9e8b87af003d550f39e9d34cf19f94088660a38)
 - [0bb740b82dcf7324dad4b1650f13838b51950368](https://github.com/magento/magento2/commit/0bb740b82dcf7324dad4b1650f13838b51950368)
 - [c51f94c3a115468ea4808478e67c88c13dca1445](https://github.com/magento/magento2/commit/c51f94c3a115468ea4808478e67c88c13dca1445)
 - [9f3f3e05d91951f56b469cf47576e3c2eb72a9b5](https://github.com/magento/magento2/commit/9f3f3e05d91951f56b469cf47576e3c2eb72a9b5)

The fix will be available with the upcoming 2.3.0 release.
